commit 3d786b91cd596b92276f69f8a5e747f552d9f759
Author: Sean Paul <seanpaul@chromium.org>
Date:   Tue Oct 30 12:00:08 2018 -0400

    drm/msm: dpu: Only check flush register against pending flushes
    
    [ Upstream commit 5f79e03b1f7c1b2cf0019ce6365fe5d52629813d ]
    
    There exists a case where a flush of a plane/dma may have been triggered
    & started from an async commit. If that plane/dma is subsequently disabled
    by the next commit, the flush register will continue to hold the flush
    bit for the disabled plane. Since the bit remains active,
    pending_kickoff_cnt will never decrement and we'll miss frame_done
    events.
    
    This patch limits the check of flush_register to include only those bits
    which have been updated with the latest commit.
    
    Changes in v2:
    - None
    
    Reviewed-by: Jeykumar Sankaran <jsanka@codeaurora.org>
    Signed-off-by: Sean Paul <seanpaul@chromium.org>
    Signed-off-by: Rob Clark <robdclark@gmail.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/gpu/drm/msm/disp/dpu1/dpu_encoder_phys_vid.c b/drivers/gpu/drm/msm/disp/dpu1/dpu_encoder_phys_vid.c
index 14fc7c2a6bb7..c9962a36b86b 100644
--- a/drivers/gpu/drm/msm/disp/dpu1/dpu_encoder_phys_vid.c
+++ b/drivers/gpu/drm/msm/disp/dpu1/dpu_encoder_phys_vid.c
@@ -331,7 +331,7 @@ static void dpu_encoder_phys_vid_vblank_irq(void *arg, int irq_idx)
 	if (hw_ctl && hw_ctl->ops.get_flush_register)
 		flush_register = hw_ctl->ops.get_flush_register(hw_ctl);
 
-	if (flush_register == 0)
+	if (!(flush_register & hw_ctl->ops.get_pending_flush(hw_ctl)))
 		new_cnt = atomic_add_unless(&phys_enc->pending_kickoff_cnt,
 				-1, 0);
 	spin_unlock_irqrestore(phys_enc->enc_spinlock, lock_flags);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1125_3d786b91cd596b92276f69f8a5e747f552d9f759.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1125_3d786b91cd596b92276f69f8a5e747f552d9f759.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
